Sources/System/FileOperations.swift

```diff
@@ -398,3 +398,51 @@ extension FileDescriptor {
   }
   #endif
 }
+
+/*System 1.2.0, @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)*/
+extension FileDescriptor {
+  #if !os(Windows)
+  /// Truncate or extend the file referenced by this file descriptor.
+  ///
+  /// - Parameters:
+  ///   - newSize: The length in bytes to resize the file to.
+  ///   - retryOnInterrupt: Whether to retry the write operation
+  ///      if it throws ``Errno/interrupted``. The default is `true`.
+  ///      Pass `false` to try only once and throw an error upon interruption.
+  ///
+  /// The file referenced by this file descriptor will by truncated (or extended) to `newSize`.
+  ///
+  /// If the current size of the file exceeds `newSize`, any extra data is discarded. If the current
+  /// size of the file is smaller than `newSize`, the file is extended and filled with zeros to the
+  /// provided size.
+  ///
+  /// This function requires that the file has been opened for writing.
+  ///
+  /// - Note: This function does not modify the current offset for any open file descriptors
+  /// associated with the file.
+  ///
+  /// The corresponding C function is `ftruncate`.
+  /*System 1.2.0, @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)*/
```

Comment:
@lorentey what's the right availability comment here?

Reply:
Yeah, I took a punt here based on what I saw with previous evolution and given there already exists a public `1.1.0` tag. LMK if it needs changing.

Reply:
@lorentey Could you advise on what this magic comment string should be? I took a look at `./Utilities/expand-availability.py` and now I think maybe it should just be the following:

```swift
/*System 1.2.0*/
```

Happy to update this to reflect your release workflow if you can point me in the right direction here.

Reply:
@lorentey LMK what this should be and I can update this PR.

Reply:
Ah, apologies for the delay -- `/*System 1.2.0*/` is going to be the right one, it just needs to be defined in `/Utilities/expand-availability.py`. (Expanded to 9999 versions.)

We can define the version directly within this PR, or we can just land it with an undefined version, and let maintainers deal with it later. (These comments do not matter within this package variant of System.)

Reply:
(To clarify, the comment we have here is just fine -- we don't need to remove the `@available` incantation.)

Reply:
Added in:

* ca6e888 `Utilities: Add version 1.2.0 to expand-availability.py tool`

Reply:
How did I miss it! 👍